Sources/Turf/Geometry.swift

```diff
@@ -4,21 +4,157 @@ import CoreLocation
+    case GeometryCollection(geometries: [Geometry])
+    
+    public var type: GeometryType {
+        switch self {
+        case .Point(_):
+            return .Point
+        case .LineString(_):
+            return .LineString
+        case .Polygon(_):
+            return .Polygon
+        case .MultiPoint(_):
+            return .MultiPoint
+        case .MultiLineString(_):
+            return .MultiLineString
+        case .MultiPolygon(_):
+            return .MultiPolygon
+        case .GeometryCollection(_):
+            return .GeometryCollection
+        }
+    }
     
-    public var geometryType: GeometryType? {
-        return GeometryType(rawValue: type)
+    public var value: Any? {
+        switch self {
+        case .Point(let value):
+            return value
+        case .LineString(let value):
+            return value
+        case .Polygon(let value):
+            return value
+        case .MultiPoint(let value):
+            return value
+        case .MultiLineString(let value):
+            return value
+        case .MultiPolygon(let value):
+            return value
+        case .GeometryCollection(let value):
+            return value
+        }
+    }
+}
+
+extension Geometry {
+    /// Returns coordinates if current enum case is `.Point`
+    public var point: CLLocationCoordinate2D? {
+        guard case let .Point(coordinates: coordinates) = self else { return nil }
+        return coordinates
     }
+    /// Returns coordinates if current enum case is `.LineString`
+    public var lineString: [CLLocationCoordinate2D]? {
+        guard case let .LineString(coordinates: coordinates) = self else { return nil }
+        return coordinates
+    }
+    /// Returns coordinates if current enum case is `.Polygon`
+    public var polygon: [[CLLocationCoordinate2D]]? {
+        guard case let .Polygon(coordinates: coordinates) = self else { return nil }
+        return coordinates
+    }
+    /// Returns coordinates if current enum case is `.MultiPoint`
+    public var multiPoint: [CLLocationCoordinate2D]? {
+        guard case let .MultiPoint(coordinates: coordinates) = self else { return nil }
+        return coordinates
+    }
+    /// Returns coordinates if current enum case is `.MultiLineString`
+    public var multiLineString: [[CLLocationCoordinate2D]]? {
+        guard case let .MultiLineString(coordinates: coordinates) = self else { return nil }
+        return coordinates
+    }
+    /// Returns coordinates if current enum case is `.MultiPolygon`
+    public var multiPolygon: [[[CLLocationCoordinate2D]]]? {
+        guard case let .MultiPolygon(coordinates: coordinates) = self else { return nil }
+        return coordinates
+    }
+    /// Returns geometries collection if current enum case is `.GeometryCollection`
+    public var geometryCollection: [Geometry]? {
+        guard case let .GeometryCollection(geometries: geometries) = self else { return nil }
+        return geometries
+    }
+}
```

Comment:
Please delete these. The idea behind an associated-value enum is to force the developer to deal with variants at point of use instead of using "magic optionals" that may or may not be nil depending on the variance.

If a developer expects a _certain_ case in an enum, they can quickly resolve for it by using `guard case let .GeometryCondition(coordinates) = enum else { panic() }`
```suggestion

```